packages/process/src/node/terminal-process.ts

```diff
@@ -8,60 +8,75 @@
+import * as pty from 'node-pty';
+import { ITerminal } from 'node-pty/lib/interfaces';
 
 export const TerminalProcessOptions = Symbol("TerminalProcessOptions");
 export interface TerminalProcessOptions {
-    command: string,
-    args?: string[],
-    options?: object
+    readonly command: string,
+    readonly args?: string[],
+    readonly options?: object
 }
 
 export const TerminalProcessFactory = Symbol("TerminalProcessFactory");
-export type TerminalProcessFactory = (options: TerminalProcessOptions) => TerminalProcess;
+export interface TerminalProcessFactory {
+    (options: TerminalProcessOptions): TerminalProcess;
+}
 
 /* Use this instead of the node-pty stream, since the node-pty stream is already resumed.  */
-export class TerminalReadableStream extends stream.Readable {
-    constructor(protected readonly terminal: any, opts?: any) {
+class ReadableTerminalStream extends stream.Readable {
+
+    constructor(protected readonly terminal: ITerminal, opts?: stream.ReadableOptions) {
         super(opts);
-        this.terminal.on('data', (data: any) => {
-            this.push(data);
-        });
+        this.terminal.on('data', data => this.push(data));
     }
+
     /* This needs to be implemented as per node's API doc, even if it's empty.  */
     _read(size: number) {
     }
+
+}
+
+class WritableTerminalStream extends stream.Writable {
+
+    constructor(protected readonly terminal: ITerminal) {
+        super({
+            write: (chunk, encoding, next) => {
+                this.terminal.write(chunk.toString());
+                next();
+            }
+        });
+    }
+
 }
 
 @injectable()
 export class TerminalProcess extends Process {
 
-    readonly type: 'Raw' | 'Terminal' = 'Terminal';
-    output: TerminalReadableStream;
-    protected process = undefined;
-    protected terminal: any;
-    protected terminalReadStream: TerminalReadableStream;
+    readonly input: stream.Writable;
+    readonly output: stream.Readable;
+    protected readonly terminal: ITerminal;
 
     constructor(
         @inject(TerminalProcessOptions) options: TerminalProcessOptions,
         @inject(ProcessManager) processManager: ProcessManager,
         @inject(ILogger) logger: ILogger) {
-        super(processManager, logger);
+        super(processManager, logger, ProcessType.Terminal);
 
         this.logger.debug(`Starting terminal process: ${options.command},`
             + ` with args : ${options.args}, `
-            + ` options ${JSON.stringify(options.options)} `);
+            + ` options ${JSON.stringify(options.options)}`);
 
         this.terminal = pty.spawn(
             options.command,
             options.args,
             options.options);
 
         this.terminal.on('exit', this.emitOnExit.bind(this));
-        this.output = new TerminalReadableStream(this.terminal);
+        this.output = new ReadableTerminalStream(this.terminal);
+        this.input = new WritableTerminalStream(this.terminal);
```

Comment:
Remove this.input, the to write to a term the one should use the write method directly or we could make a stream over write...